apis/core/v1alpha2/readinessprovider_webhook.go

```diff
@@ -46,13 +54,39 @@ func (r *ReadinessProvider) ValidateDelete() error {
 
 func (r *ReadinessProvider) validateObject() error {
 	var allErrors field.ErrorList
+	specPath := field.NewPath("spec")
+
+	saSource := r.Spec.ServiceAccount
+	if saSource != nil {
+		if saSource.Name == "" {
+			allErrors = append(allErrors, field.Required(specPath.Child("serviceAccount").Child("Name"), "missing required field"))
+		}
+
+		if saSource.Namespace == "" {
+			allErrors = append(allErrors, field.Required(specPath.Child("serviceAccount").Child("Namespace"), "missing required field"))
+		}
+	}
+
+	// Checking if provided serviceaccount is valid
+	if saSource != nil && len(allErrors) == 0 {
+		sa := &corev1.ServiceAccount{}
+		readinessproviderlog.Info("checking if service account is present", "source", saSource)
+		err := cli.Get(context.Background(), client.ObjectKey{
```

Comment:
Maybe a create a context in calling functions and pass it to `validateObject`?

Reply:
Done